purge.go

```diff
@@ -0,0 +1,44 @@
+package lock
+
+// A Purger deletes expired locks.
+type Purger interface {
+	// Purge deletes expired locks. It is important to note that not just
+	// locks with a TTL of 0 get deleted - any lock that shares a lockId
+	// with a lock that has has a TTL of 0 will be removed.
+	Purge() ([]LockStatus, error)
+}
+
+// purge implements the Purger interface.
+type purger struct {
+	locker Locker
+}
+
+// NewPurger creates a new Purger.
+func NewPurger(locker Locker) Purger {
+	return &purger{
+		locker: locker,
+	}
+}
+
+func (p *purger) Purge() ([]LockStatus, error) {
+	// Get all locks with a TTL of less than 1, i.e. 0.
+	filter := Filter{
+		TTLlt: 1,
+	}
+	locks, err := p.locker.Status(filter)
```

Comment:
Is it possible that we have a "race condition" where:
1. Lock A is selected for purging
2. Lock A is renewed to have another TTL in the future.
3. Lock A is purged

Reply:
ah yeah, good catch. I'll fix that

Reply:
Fixed by only allowing locks to be renewed if their TTL is at least 1 second. This gets ride of the potential race since only locks with a TTL of less than 1 second (i.e., 0 seconds) can be deleted by the Purger